smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java

```diff
@@ -345,19 +345,32 @@ private String getActualAddress(Message<?> message, io.vertx.mutiny.amqp.AmqpMes
                     .contentType(JSON_CONTENT_TYPE);
         }
 
-        builder.address(metadata.map(OutgoingAmqpMetadata::getAddress).orElse(null));
-        builder.applicationProperties(metadata.map(OutgoingAmqpMetadata::getProperties).orElseGet(JsonObject::new));
-
-        builder.contentEncoding(metadata.map(OutgoingAmqpMetadata::getContentEncoding).orElse(null));
-        builder.contentType(metadata.map(OutgoingAmqpMetadata::getContentType).orElse(null));
-        builder.correlationId(metadata.map(OutgoingAmqpMetadata::getCorrelationId).orElse(null));
-        builder.groupId(metadata.map(OutgoingAmqpMetadata::getGroupId).orElse(null));
-        builder.id(metadata.map(OutgoingAmqpMetadata::getId).orElse(null));
-        int priority = metadata.map(OutgoingAmqpMetadata::getPriority).orElse(-1);
-        if (priority >= 0) {
-            builder.priority((short) priority);
-        }
-        builder.subject(metadata.map(OutgoingAmqpMetadata::getSubject).orElse(null));
+        metadata.ifPresent(meta -> {
```

Comment:
Any chance you could avoid the lambda? It doesn't bring much here so better not pay for its cost.

Reply:
Reactive MEssaging and Reactive is general uses lots of lambdas. I can remove this one, but there will be tons of them around.

Reply:
Done.

Reply:
Yeah, maybe it's just me but I fail to see how a lambda is better than a simple `if (optional.isPresent())`.

I can see how it gets useful in more complex cases but I would try to avoid them for the simple ones.

Anyway, ignore me, I'm just ranting :).